pkg/apis/vsphereprovider/v1alpha1/vsphereproviderconfig_types.go

```diff
@@ -52,8 +52,38 @@ type VSphereMachineProviderSpec struct {
 	// machine is cloned.
 	// +optional
 	DiskGiB int32 `json:"diskGiB,omitempty"`
+	// Snapshot is the name of the snapshot from which the VM was cloned
+	// +optional
+	Snapshot string `json:"snapshot"`
+
+	// CloneMode specifies the type of clone operation.
+	// The LinkedClone mode is only support for templates that have at least
+	// one snapshot. If the template has no snapshots, then CloneMode defaults
+	// to FullClone.
```

Comment:
Does it not default to LinkedClone based on https://github.com/openshift/machine-api-operator/blob/913ee8295fc4b39926cbafa708fa97b67dc8c975/pkg/controller/vsphere/reconciler.go#L382?

Reply:
It defaults here https://github.com/openshift/machine-api-operator/pull/512/files#diff-11b6610f82e5d47fdfb8dda44b90dd1eR405 when no snapshots are found